homeassistant/components/atag/sensor.py

```diff
@@ -36,37 +36,27 @@ class AtagSensor(AtagEntity, SensorEntity):
     def __init__(self, coordinator, sensor):
         """Initialize Atag sensor."""
         super().__init__(coordinator, SENSORS[sensor])
-        self._name = sensor
-
-    @property
-    def state(self):
-        """Return the state of the sensor."""
-        return self.coordinator.data.report[self._id].state
-
-    @property
-    def icon(self):
-        """Return icon."""
-        return self.coordinator.data.report[self._id].icon
-
-    @property
-    def device_class(self):
-        """Return deviceclass."""
-        if self.coordinator.data.report[self._id].sensorclass in [
+        self._attr_name = sensor
+        self._attr_icon = coordinator.data.report[self._id].icon
```

Comment:
The icon is provided by the upstream API, does this icon reflect the current state or not? I'm not sure, have you tested it?

Reply:
From looking into `pyatag`, it seems like some sensors depend on state. So this is reverted.